python/ray/experimental/async_plasma.py

```diff
@@ -0,0 +1,235 @@
+import asyncio
+import sys
+
+import pyarrow.plasma as plasma
+
+import ray
+from ray.services import logger
+
+
+def _release_waiter(waiter, *_):
+    if not waiter.done():
+        waiter.set_result(None)
+
+
+class PlasmaProtocol(asyncio.Protocol):
+    """
+    Protocol control for the asyncio connection.
+    """
+
+    def __init__(self, plasma_client, plasma_event_handler):
+        self.plasma_client = plasma_client
+        self.plasma_event_handler = plasma_event_handler
+        self.transport = None
+        self._buffer = b""
+
+    def connection_made(self, transport):
+        self.transport = transport
+
+    def data_received(self, data):
+        self._buffer += data
+        messages = []
+        i = 0
+        while i + 8 <= len(self._buffer):
```

Comment:
What are these `8` values? This needs to be documented clearly and probably use a variable with a clear name instead of `8`